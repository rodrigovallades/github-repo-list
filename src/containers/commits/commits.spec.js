import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import { Commits } from './index'

jest.mock('react-router-dom', () => ({ Link: 'Link' }))

describe('Commits container', function () {
  beforeEach(function() {
    global.sessionStorage = jest.genMockFunction();
    global.sessionStorage.setItem = jest.genMockFunction();
    global.sessionStorage.getItem = jest.genMockFunction();
  });

  it('renders "Commits" deep correctly', () => {
    const tree = renderer.create(<Commits/>);
    expect(tree).toMatchSnapshot();
  });

  it('renders "Commits" shallow correctly', () => {
    const component = shallow(<Commits/>);
    expect(component.exists()).toEqual(true);
  });
})
