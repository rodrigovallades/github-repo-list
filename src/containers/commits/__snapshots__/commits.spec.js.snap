// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Commits container renders "Commits" deep correctly 1`] = `
<div>
  <div
    className="container"
  >
    <h1
      className="title"
    >
      <span
        className="float-right"
      >
        <small>
          <Link
            to="/repos"
          >
            &lt; repos
          </Link>
        </small>
      </span>
       
      <span
        className="badge badge-light"
      >
        0
      </span>
       
       
      <small
        className="text-muted"
      >
        commits
      </small>
    </h1>
    <input
      className="filter form-control"
      onChange={[Function]}
      placeholder="Filter commits"
      type="text"
      value=""
    />
    <div
      className="list-group commits"
    >
      <p>
        No commits found in this repository.
      </p>
    </div>
    <div
      className="alert alert-info"
      role="alert"
    >
      <span
        className="octicon octicon-info"
      />
       Tip: Download the 
      <strong>
        <a
          href="https://github.com/zalmoxisus/redux-devtools-extension"
          target="_blnk"
        >
          Redux DevTools
        </a>
      </strong>
       to inspect the Redux store state.
    </div>
  </div>
</div>
`;
